Firmware/targets/arm/Src/encoder_hw.cpp

```diff
@@ -0,0 +1,129 @@
+#include "encoder_hw.h"
+
+#include "stm32f1xx_hal_gpio.h"
+#include "stm32f1xx_hal_tim.h"
+#include "tim.h"
+
+#define DEBOUNCE_STABLE 5
```

Comment:
extract constants to another file; preffered use constexpr static in C++
(config file have to be in C++ format because of rest of the code)